src/ws/shard.ts

```diff
@@ -327,7 +328,7 @@ async function processGatewayQueue() {
     return;
   }
 
-  basicShards.forEach((shard) => {
+  await Promise.all(basicShards.map(async (shard) => {
```

Comment:
This is unnecessary, since we are not really returning a promise from the .map()

Reply:
It's actually needed since to handle the rate limit we need to wait until everything is done and then wait 2 seconds if I'm not wrong.